dspace-spring-rest/src/test/java/org/dspace/app/rest/BitstreamFormatRestRepositoryIT.java

```diff
@@ -10,89 +10,485 @@
+
+        int nonExistentBitstreamFormatID = 404404404;
+
+        BitstreamFormatRest bitstreamFormatRest = converter.fromModel(bitstreamFormat);
+        String token = getAuthToken(admin.getEmail(), password);
+        //Update it with non existent ID in URL and in JSON
+        bitstreamFormatRest.setShortDescription("Test short UPDATED");
+        bitstreamFormatRest.setId(nonExistentBitstreamFormatID);
+        getClient(token).perform(put("/api/core/bitstreamformats/" + nonExistentBitstreamFormatID)
+                .content(mapper.writeValueAsBytes(bitstreamFormatRest)).contentType(contentType))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void updateNonExistingIDInJustURL() throws Exception {
+        ObjectMapper mapper = new ObjectMapper();
+        //Create bitstream format
+        context.turnOffAuthorisationSystem();
+        BitstreamFormat bitstreamFormat = BitstreamFormatBuilder.createBitstreamFormat(context)
+                .withMimeType("application/octet-stream")
+                .withDescription("Description - updateNonExistingIDInJustURL")
+                .build();
+        context.restoreAuthSystemState();
+
+        int nonExistentBitstreamFormatID = 404404404;
+
+        BitstreamFormatRest bitstreamFormatRest = converter.fromModel(bitstreamFormat);
+        String token = getAuthToken(admin.getEmail(), password);
+        //Update it with non existent ID in URL
+        bitstreamFormatRest.setShortDescription("Test short UPDATED");
+        getClient(token).perform(put("/api/core/bitstreamformats/" + nonExistentBitstreamFormatID)
+                .content(mapper.writeValueAsBytes(bitstreamFormatRest)).contentType(contentType))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void updateNonExistingIDInJSONButValidInURL() throws Exception {
+        ObjectMapper mapper = new ObjectMapper();
+        //Create bitstream format
+        context.turnOffAuthorisationSystem();
+        BitstreamFormat bitstreamFormat = BitstreamFormatBuilder.createBitstreamFormat(context)
+                .withMimeType("application/octet-stream")
+                .withDescription("Description - updateNonExistingIDInJSONButValidInURL")
+                .build();
+        context.restoreAuthSystemState();
+
+        int nonExistentBitstreamFormatID = 404404404;
+
+        BitstreamFormatRest bitstreamFormatRest = converter.fromModel(bitstreamFormat);
+        String token = getAuthToken(admin.getEmail(), password);
+        //Update it with non existent ID in JSON, but valid in URL
+        bitstreamFormatRest.setShortDescription("Test short UPDATED");
+        bitstreamFormatRest.setId(nonExistentBitstreamFormatID);
+        getClient(token).perform(put("/api/core/bitstreamformats/" + bitstreamFormat.getID())
+                .content(mapper.writeValueAsBytes(bitstreamFormatRest)).contentType(contentType))
+                .andExpect(status().isBadRequest());
```

Comment:
This has been resolved. The discussion on changing the status code can indeed best be delayed until a later PR since the merged rest contract also doesn't mention it